### DISCUSSION TITLE
ClientResponseError 0: ClientResponseError

### DISCUSSION THREAD

**Discussion by SebberJJ** [pocketbase/pocketbase#5162] [Q&A] [answered]
Hi

I have a weird problem. I am trying to get the data inside the collection: "servers" - I am getting my data, but i also get an error in my console: 
`Single-Server-Table.jsx:22 Error occurred while trying to get the server list ClientResponseError 0: ClientResponseError
    at pocketbase.js?v=cbba0b48:889:13
    at async getServerList (Single-Server-Table.jsx:11:25)`
    
If i use the same code another place in my project, i get the same error twice and now none of my data is showing? If i use the code in one place the data is showing with one error?

No logs with any errors in pocketbase

My code;

```
const [serverList, setServerList] = useState([])
  const [loading, setLoading] = useState(true)
  
  useEffect(() => {
    const getServerList = async () => {
      try {
        const servers = await pb.collection('servers').getList()
        console.log(servers)
        const mapServers = servers.items.map((items) => ({
          id: items.id,
          name: items.server_name,
          ip: items.ip_and_port,
          logo: items.server_logo,
          banner: items.server_banner,
        }))
        setServerList(mapServers)
      } catch (error) {
        console.error("Error occurred while trying to get the server list", error)
      } finally {
        setLoading(false)
      }
    }
    getServerList()
  }, [])
```
This is my collection

[pb_schema.json](https://github.com/user-attachments/files/16086398/pb_schema.json)

  
  I use ViteReact with Javascript

**Comment by ganigeorgiev** [comment, marked as answer]
`ClientResponseError 0` usually indicates a client-side or network connectivity error (aka. the request was interrupted and has never reached the server, which is also the reason why there are no logs in the PocketBase Admin UI).

> If i use the same code another place in my project, i get the same error twice and now none of my data is showing? If i use the code in one place the data is showing with one error?

It is possible that you are hitting the autocancellation - https://github.com/pocketbase/js-sdk?tab=readme-ov-file#auto-cancellation. If you don't want this behavior, you can disable it globally by calling `pb.autoCancellation(false)` after the SDK initialization.

For more details you could also inspect `console.log(error.originalError)`.

**Reply by SebberJJ** [reply]
> `ClientResponseError 0` usually indicates a client-side or network connectivity error (aka. the request was interrupted and has never reached the server, which is also the reason why there are no logs in the PocketBase Admin UI).
> 
> > If i use the same code another place in my project, i get the same error twice and now none of my data is showing? If i use the code in one place the data is showing with one error?
> 
> It is possible that you are hitting the autocancellation - https://github.com/pocketbase/js-sdk?tab=readme-ov-file#auto-cancellation. If you don't want this behavior, you can disable it globally by calling `pb.autoCancellation(false)` after the SDK initialization.
> 
> For more details you could also inspect `console.log(error.originalError)`.

I used `pb.autoCancellation(false)` and i can now use the code multiple places and not only that it also got rid of my error? 

Will this give me any problems in the future with using `pb.autoCancellation(false)` ? - Or is this a good way of fixing the error? :)

**Reply by ganigeorgiev** [reply]
It depends on your use case. You can disable it also on per request basis if you specify the `requestKey` as shown in the linked docs.

Usually if the above code is executed on the client-side (browser, mobile device, etc.) having auto cancellation enabled can help identify and prevent async bugs in your code. For example, if you are sending a request after a button click, in case the user spam the button you usually want only the last click to resolve and all the previous pending requests to be ignored, something like:

```js
async function onClick() {
    try {
        const result = await pb.collection("example").getList(1, 30)
        
        ...

    } catch (err) {
        // ignore abort errors, otherwise show the error message
        if (!err.isAbort) {
            window.alert(err.response.message)
        }
    }
}
```


If you are using the JS SDK as admin client in a server-context, then the auto cancellation needs to be disabled because in this case you have a single JS SDK instance that needs to be able to resolve requests from multiple different users (eg. the example in https://github.com/pocketbase/pocketbase/discussions/3792#discussioncomment-7621130).

**Comment by Sebasmzt** [comment]
This only happends in local, but is not showing any error in production

**Reply by ganigeorgiev** [reply]
Please don't hijack resolved discussions. If you need further help, open a new Q&A discussion elaborating more on your problem and apps setup to understand what is the issue in the first place.
